Update daily workflow to sharding

lanej0 opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Updates the daily testing workflows to execute end-to-end tests with sharding.

This PR also makes a fix to the reporting -- skipped tests have been reported higher than they should have because the Playwright blob reporter and allure reporters were conflicting.  This PR will correct that for all runs.

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to Actions > Daily smoke test
2. Run the workflow using this branch ([here](https://github.com/woocommerce/woocommerce/actions/runs/7807131110) is a sample run if you don't want to wait).
3. Ensure workflow passes -- k6 tests are failing on daily currently, so they fail here too.  We'll look at fixing that in a different PR.

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -->

</details>

Commit:
Update daily workflow to sharding

Commit:
Changelog

Commit:
Fix syntax

Commit:
Fix check

Commit:
Shard plugin tests

Commit:
Can’t use matrix in composite run action

Commit:
Merge branch 'trunk' into dev/update-daily-e2e-to-sharding

Commit:
Remove blob reporting

Commit:
Update publish step

adimoldovan commented on .github/workflows/smoke-test-daily.yml:
```diff
@@ -117,22 +128,67 @@ jobs:
+                  retention-days: 1
+                  compression-level: 9
+            
+    e2e-tests-success:
```
I guess this job is not really needed, this workflow doesn't block anything being a scheduled one.

lanej0 replied:
Right. I was copying and pasting from the PR workflow :-)

adimoldovan commented on .github/workflows/smoke-test-daily.yml:
```diff
@@ -117,22 +128,67 @@ jobs:
+              run-id: e2e-tests-run
+              merge-multiple: true
+
+        - name: Generate Playwright E2E Test report.
```
Nitpick, but `Generate Playwright E2E Test report.` makes me think the Playwright default report will be generated here, while in fact the Allure report is generated. If there was no Playwright report available it wouldn't be confusing. How about keeping it simple: `Generate test report`?

adimoldovan commented on .github/workflows/smoke-test-daily.yml:
```diff
@@ -117,22 +128,67 @@ jobs:
+    merge-reports:
+      name: Merge e2e test reports
+      if: always()
+      needs: e2e-tests-success
```
The `e2e-tests-success` only runs on `success()`, so I guess this one will not run on tests failures.
But I think `e2e-tests-success` is not needed in this case (see my other comment) and this job should wait for `e2e-tests` instead.

Commit:
Feedback from review

Commit:
Fix WooPayments post

lanej0 commented:
Thanks for your review @adimoldovan -- I've implemented all of your feedback, and fixed the last couple of bugs that were lingering with reporting.  The latest run can be seen here: https://github.com/woocommerce/woocommerce/actions/runs/7821375862

adimoldovan reviewed (approved):
Looks good 👍🏼
